### PR TITLE
chore(ui): Update browserslist to Chrome 80 in ui

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

TL;DR 10% decrease for main.js bundle and 7.5% for total of js bundle files.

Following recent theme of deleting css, media, and js files, reduce size of bundle via modern JavaScript.

### Problems

1. Thanks to Brad for demo of Lighthouse browser tool that reported that bundles do not use modern JavaScript.

2. Also `yarn build` warns several times:

    > Browserslist: caniuse-lite is outdated. Please run:
    > npx update-browserslist-db@latest

### Analysis

1. Default browserslist query in package.json file provoked warnings about caniuse-lite outdated because of relative conditions:

    ```
    "browserslist": [
        ">0.2%",
        "not dead",
        "not ie <= 11",
        "not op_mini all"
    ]
    ```

2. For determinate results similar to what yarn.lock does for package.json file, change to specify a specific browser version `"Chrome >= 80"` that has ECMAScript 2020 features we use:

    https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#browser_compatibility

    > The nullish coalescing `??` operator is a logical operator that returns its right-hand side operand when its left-hand side operand is `null` or `undefined`, and otherwise returns its left-hand side operand.

    https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility

    > The optional chaining `?.` operator accesses an object's property or calls a function. If the object accessed or function called using this operator is `undefined` or `null`, the expression short circuits and evaluates to undefined instead of throwing an error.

### Lessons

First attempt was to update `"target": "es5"` property in tsconfig.json but it did not make any difference.

https://babeljs.io/docs/babel-plugin-transform-typescript#typescript-compiler-options

> The official TypeScript compiler has many options for configuring how it compiles and type checks. While many don't apply, some behaviors might be useful and their equivalents in Babel can be enabled by some configuration options or plugins.

> `--target` Babel doesn't support targeting a specific version of the language, but you can choose which engines you want to target using `@babel/preset-env`.

https://babeljs.io/docs/babel-preset-env

> `@babel/preset-env` is a smart preset that allows you to use the latest JavaScript without needing to micromanage which syntax transforms (and optionally, browser polyfills) are needed by your target environment(s). This both makes your life easier and JavaScript bundles smaller!

> We leverage these data sources to maintain mappings of which version of our supported target environments gained support of a JavaScript syntax or browser feature, as well as a mapping of those syntaxes and features to Babel transform plugins and core-js polyfills.

### Residue

Investigate pro and con to intentionally update babel package dependencies and how not to make a mess of yarn.lock file.

1. Initial update to `@babel/core` caused too many changes in yarn.lock file and did not affect the build sizes.

2. While investigating item 1, I noticed `@babel/runtime` package.

    https://babeljs.io/docs/babel-runtime

    > Sometimes Babel may inject some code in the output that is the same across files, and thus can be potentially re-used.

    > With `@babel/plugin-transform-runtime`, it would replace the reference to the function to the @babel/runtime version.

    > `@babel/runtime` is just the package that contains the implementations of the functions in a modular way.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

1. Update ui/apps/platform/package.json file, except for `npx` which updates yarn.lock file directly
2. `yarn` in ui, except for `npx`
3. `yarn build` in ui
4. `wc build/static/js/*.js` in ui

|                                  | main.js | total js |
| :---                             |    ---: |     ---: |
| baseline                         | 4697789 | 11153046 |
| `"Chrome >= 80"`                 | 4238134 | 10318844 |
| `npx update-browserslist@latest` | 4238134 | 10318844 |
| improved - baseline              | -459655 |  -834202 |
| improved / baseline              |   0.902 |    0.925 |

### Interpretation

1. Besides being indeterminate even if caniuse is kept up-to-date, `>0.2%` query for outdated data was less effective than determinate `Chrome >= 80` even before update to caniuse.

2. If most language features that we use are supported by the browser version that we specify, it makes sense that update to core-js did not decrease bundle size beyond the first reduction.

    https://www.npmjs.com/package/core-js

    > Modular standard library for JavaScript. Includes polyfills for ECMAScript up to 2023: promises, symbols, collections, iterators, typed arrays, many other features, ECMAScript proposals, some cross-platform WHATWG / W3C features and proposals like URL. You can load only required features or use it without global namespace pollution.

### CODE REVIEW

**Review by vjwilson — APPROVED** [review]
Are there benefits to adding Firefox or Safari to this list?

**Comment by pedrottimark** [comment]
Good question. No, babel-preset-env needs to know which language features are supported or not.

We can use any one of the browsers as the means to that end. More is not better, but more complicated.

For your info, equivalent versions for nullish coalescing and optional chaining:
* Firefox 72
* Safari 13.1
